src/posix/net.cpp

```diff
@@ -365,76 +460,74 @@ void CurlHttpIO::filterDNSservers()
 void CurlHttpIO::addaresevents(Waiter *waiter)
 {
     CodeCounter::ScopeTimer ccst(countAddAresEventsCode);
-    closearesevents();
+
+    SockInfoMap prevAressockets;   // if there are SockInfo records that were in use, and won't be anymore, they will be deleted with this
+    prevAressockets.swap(aressockets);
```

Comment:
thanks for the thoughtful analysis here.   For win32, it's much easier to reuse the same data structure and technique for c-ares as for c-url.   For linux, I thought the balance was that it would be simpler to maintain the same code for both platforms as much as possible.   Since DNS lookups should be cached most of the time, the difference between map and vector lookup here should be negligible since it will only be used occasionally.   thanks!